### PR TITLE
Provided the ability to pass the proxy to the runtime

### PR DESCRIPTION
**Pull request by SamvelRaja**
```js
...
 runner.run(collection, {
                requester: {
                    proxyManager: ProxyManager,
                }
            }
})
...
```

where,

```js
/**
*  @params url  [Postman Request Url](http://www.postmanlabs.com/postman-collection/Url.html)
*  @params cb  function
*/
ProxyManager.getProxyConfiguration = function (url, cb) {
        return cb(null, {proxy: proxyURL, tunnel: false});
}
```

### CODE REVIEW

File: lib/requester/request-wrapper.js [hunk 1]
```diff
@@ -61,36 +61,85 @@ var _ = require('lodash'),
...
+            options.lookup = _.isFinite(portNumber) && lookup.bind(this, portNumber);
+        }
+
+        if (!isSSL || !(options.certificateManager && options.certificateManager.getCertificateContents)) {
+            return cb(null, options);
+        }
+
+        // Get any certificates for this request from the certificate manager
+        options.certificateManager.getCertificateContents(hostname + ':' + port,
+              function (err, info) {
```
Comment (shamasis) [inline lib/requester/request-wrapper.js]: fix indentation

File: lib/requester/request-wrapper.js [hunk 2]
```diff
@@ -61,36 +61,85 @@ var _ = require('lodash'),
...
+            portNumber = _.parseInt(portNumber, 10);
+            options.lookup = _.isFinite(portNumber) && lookup.bind(this, portNumber);
+        }
+
+        if (!isSSL || !(options.certificateManager && options.certificateManager.getCertificateContents)) {
+            return cb(null, options);
+        }
+
+        // Get any certificates for this request from the certificate manager
+        options.certificateManager.getCertificateContents(hostname + ':' + port,
```
Comment (shamasis) [inline lib/requester/request-wrapper.js]: Use SDK helper getRemote()

File: lib/requester/request-wrapper.js [hunk 3]
```diff
@@ -61,36 +61,85 @@ var _ = require('lodash'),
...
+    setProxy = function(options, cb) {
+        if (options.proxyManager && options.proxyManager.getProxyConfiguration) {
+            options.proxyManager.getProxyConfiguration(options.request.url, function(err, obj) {
+                options.proxy = obj && obj.proxy,
+                options.tunnel = obj && obj.tunnel;
+                if (err) {
+                  return cb(err, options);
+                }
+
+                return cb(null, options);
```
Comment (shamasis) [inline lib/requester/request-wrapper.js]: just do cb(err, options); the `if` block is redundant

File: lib/requester/request-wrapper.js [hunk 4]
```diff
@@ -61,36 +61,85 @@ var _ = require('lodash'),
...
-        function (err, info) {
-            _.extend(options, {
-                key: info && info.key,
-                cert: info && info.pem,
-                passphrase: info && info.passphrase
-            });
-            requests(options, callback);
+    var request = options.request;
+
+    setProxy(options, function(err, options) {
```
Comment (shamasis) [inline lib/requester/request-wrapper.js]: Use `async` now.

File: lib/requester/request-wrapper.js [hunk 5]
```diff
@@ -61,36 +61,85 @@ var _ = require('lodash'),
...
+     */
+    setCertificate = function (options, cb) {
+        var request = options.request,
+            // todo: ensure that the sdk trims the host
+            isSSL = _.startsWith(request.url.protocol, 'https'),
+            hostname = request.url && _.isFunction(request.url.getHost) && request.url.getHost().toLowerCase(),
+            portNumber;
+            port = _.get(request, 'url.port');
+
+        if (hostname === LOCALHOST) {
```
Comment (shamasis) [inline lib/requester/request-wrapper.js]: localhost lookup should not be part of set certificate

File: lib/requester/request-wrapper.js (hunk 3 above)
Comment (SamvelRaja) [inline lib/requester/request-wrapper.js]: yeah 👍

**Pull request merged**